### PR TITLE
refact(GlobalBanner): integrate new Banner UI componnet

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Updates `GlobalBanner` so it makes usage `Banner` component.
 
## Justify why they are needed

`GlobalBanner` is only responsible for displaying a `Banner` globally - UI is managed by `Banner` which down the stack will also be used by `AnnouncementBlock`

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2469** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2469" target="_blank">Graphite</a> 
    * **PR #2470** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2470" target="_blank">Graphite</a>  👈
      * **PR #2471** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2471" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2470?utm_source=stack-comment).

File: apps/store/src/components/GlobalBanner/GlobalBanner.tsx [hunk 1]
```diff
@@ -1,65 +1,24 @@
...
 import { useGlobalBanner, useGlobalBannerClosed } from './useGlobalBanner'
 
 export const GlobalBanner = () => {
-  const [value, setValue] = useGlobalBanner()
+  const [banner, setBanner] = useGlobalBanner()
   const [closed, setIsClosed] = useGlobalBannerClosed()
 
   useDebugShopSession()
 
   const handleClose = () => {
```
Comment (alebedev) [inline apps/store/src/components/GlobalBanner/GlobalBanner.tsx:12]: Is there a way to store banner and closed in the same state element?  It looks odd that they are unrelated and have to changed independently

**Review by robinandeer — APPROVED** [review]
Other than @alebedev comment -- I think it looks good

File: apps/store/src/components/GlobalBanner/GlobalBanner.tsx (hunk 1 above)
Comment (guilhermespopolin) [reply apps/store/src/components/GlobalBanner/GlobalBanner.tsx:12]: They need to be separated things because of the way `useDiscountBanner` works. That hook will check for check for redeemed campaigns within the shop session **and** check if the user, during his session, haven't dismissed the banner and on that scenario add a banner informing the about campaign code. If I keep them together it will be very hard to check if I should add a banner about campaign code or not.

Comment (alebedev) [reply apps/store/src/components/GlobalBanner/GlobalBanner.tsx:12]: As an idea, look at https://jotai.org/docs/guides/composing-atoms and derived atoms.  I think we're dealing with something that can be described as a value derived from 2 different inputs.  Let's try to express it via library API

Comment (guilhermespopolin) [reply apps/store/src/components/GlobalBanner/GlobalBanner.tsx:10]: I've updated the PR with derivated atoms. Please, give a look :)

**Review by alebedev — APPROVED** [review]
